docs/2_BEST_PRACTICES.md

```diff
@@ -0,0 +1,161 @@
+# Best Practices
+
+Here you can read about how we build our React applications.
+
+## Code Style
+
+Use [ESLint](http://eslint.org/) to enforce a code style. The code style
+should be consistent across each project. We provide a [default style](../.eslintrc),
+but each team is free to customize it as they see fit.
+
+## React
+
+[React](https://facebook.github.io/react/) is a library from Facebook for
+building user interfaces.
+
+### Components
+
+Use stateless (or functional) components whenever possible. Class components
+are only necessary when you need a local state, e.g. input fields.
+
+**Do**
+
+```javascript
+const Hello = ({message}) =>
+  <h1>Hello World</h1>
+```
+
+**Do NOT do**
+
+```javascript
+class Hello extends Component {
+  render() {
+    return <h1>Hello World</h1>
+  }
+}
+```
+
+### Higher-Order Components
+
+Use [higher-order components](https://facebook.github.io/react/docs/higher-order-components.html)
+(or decorators) to reuse component logic. A higher-order component is a function
+that takes a component and returns a new component.
+
+### Context
+
+Use [context](https://facebook.github.io/react/docs/context.html) together with
+higher-order components to pass props down the component tree when necessary.
+Context should be avoided when possible, because it makes the application more
+complex and harder to understand.
+
+## State
+
+Use [Redux](http://redux.js.org/) to manage the application state (also known
+as root state).
+
+### Reducers
+
+You should try to keep your application state as flat as possible.
+
+**Do**
+
+```javascript
+const todosReducer = (state = [], action) => {
+  switch (action.type) {
+    case 'ADD_TODO':
+      return [...state, action.payload]
+  }
+}
+```
+
+**Do NOT do**
+
+```javascript
+const todoReducer = (state = {todos: []}, action) => {
+  switch (action.type) {
+    case 'ADD_TODO':
+      return {
+        ...state,
+        todos: [...state.todos, action.payload],
```

Comment:
Well, this could be debated but due to the fact that we want less lines in our diffs we have opted to add a trailing comma to some projects on the last line in multi-line arrays and objects.